File: task1/task1.rs
use std::io;

fn isPrime(n: u32) -> bool {
    if n <= 1 {
        return false;
    }
    for i in 2..=((n as f64).sqrt() as u32) {
        if n % i == 0 {
            return false;
        }
    }
    true
}

fn main() {
    println!("Введите основание:");
    let mut a = String::new();
    io::stdin().read_line(&mut a).expect("Ошибка при чтении ввода.");
    let a: u32 = a.trim().parse().expect("Введено некорректное значение.");

    println!("Введите модуль:");
    let mut p = String::new();
    io::stdin().read_line(&mut p).expect("Ошибка при чтении ввода.");
    let p: u32 = p.trim().parse().expect("Введено некорректное значение.");

    println!("Введите степень:");
    let mut degree = String::new();
    io::stdin().read_line(&mut degree).expect("Ошибка при чтении ввода.");
    let degree: u32 = degree.trim().parse().expect("Введено некорректное значение.");

    let mut result = 1;

    if isPrime(p) {
        if a % p == 0 {
            println!("Условие теоремы Ферма не выполнено!");
        } else {
            for _ in 1..= degree {
                result = (result * a) % p;
            }
            println!("Результат: { }", result);
        }
    } else {
        println!("Модуль не является простым числом!");
    }
}

File: task1/task1.cpp
#include <iostream>
#include <cmath>
using namespace std;

bool isPrime(int n) //функция определения простого числа
{
    for (int i = 2; i <= sqrt(n); i++)
    {
        if (n % i == 0)
            return false;
    }
    return true;
}

int main() {
    setlocale(LC_ALL, "Russian");
    cout << "Введите основание: ";
    int a;
    cin >> a;
    cout << "Введите модуль: ";
    int p;
    cin >> p;
    cout << "Введите степень: ";
    int degree;
    cin >> degree;
    int result = 1; //искомое число, начинаем с единицы, чтобы работал цикл

    if (isPrime(p)) { //проверка модуля на простоту
        if (a % p == 0) { //проверка условия Ферма
            cout << "Условие теоремы Ферма не выполнено!";
        } else {
            for (int i = 1; i <= degree; i++) { //вычисляем число
                result = (result * a) % p;
            }
            cout << "Результат: " << result;
        }
    } else {
        cout << "Модуль не является простым числом!";
    }
    return 0;
}
